docs/blueprints/EP038.rst

```diff
@@ -0,0 +1,151 @@
+:EP: 38
+:Title: Broadcasting events
+:Authors:
+- Luis;
+- Italo;
+- Vinicius;
+- Jeronimo Bezerra jbezerra@fiu.edu;
+:Issued Date: to be defined
+:Status: Pending
+:Type: Standards Track
+
+EP03X - Broadcasting events
+
+########
+Abstract
+########
+
+This blueprint details how Kytos should export/broadcast its events
```

Comment:
This NApp is responsible for two key functions:

Publishing Serialized KytosEvents to a Message Broker:

The primary responsibility of this NApp is to publish serialized KytosEvents to a message broker. Serialized KytosEvents contain structured data representing events within the Kytos network infrastructure.
This involves converting KytosEvents into a serialized format compatible with the message broker's communication protocol.
The NApp ensures the reliable delivery of these serialized events to the message broker, handling any necessary error handling or retry mechanisms.

Filtering KytosEvents for Publication:

Additionally, this NApp is responsible for filtering KytosEvents before publishing them to the message broker.
Filtering may involve selecting specific types of events, applying rules or criteria to determine which events are relevant for publication, or transforming events into a different format as downstream consumers require.
The NApp optimizes communication by only transmitting relevant data to the message broker by filtering events, reducing unnecessary network traffic and processing overhead.

Overall, this NApp acts as a bridge between Kytos' internal message bus and the message broker, facilitating the exchange of events between the internal network infrastructure and external systems. It guarantees the serialization, filtering, and reliable transmission of relevant events to the message broker for downstream applications or service consumption.

Reply:
2.- mq_producer NApp
The mq_producer NApp should support parameterizing the exchange and routing key for publishing a given set of events. This capability allows for precise control over the distribution of events in the message broker's infrastructure.

This topic exchange operates similarly to a direct exchange but introduces a more adaptable routing mechanism based on routing patterns. Unlike direct exchanges, which rely on fixed routing keys, topic exchanges utilize wildcards for message routing, enhancing flexibility.

Here's a breakdown of its operation:

Instead of depending on specific routing keys, topic exchanges route messages by comparing a message's routing key with predefined patterns.

This comparison determines the routing of messages to one or more queues based on their correspondence with the specified patterns.

The routing key consists of a series of words separated by periods (".").

In summary, the topic exchange facilitates nuanced and dynamic message routing through wildcard patterns, offering increased flexibility and versatility in distributing messages within the messaging system.